### DISCUSSION TITLE
ModuleNotFoundError: module 'pghistory' has no attribute 'UpdateEvent'

### DISCUSSION THREAD

**Discussion by v-anne** [freelawproject/courtlistener#4524] [Q&A] [answered]
I'm attempting to run CourtListener locally and am running into the following error that is forcing the cl-django and cl-celery containers to exit prematurely:
`AttributeError: module 'pghistory' has no attribute 'UpdateEvent'`
`ModuleNotFoundError: module 'pghistory' has no attribute 'UpdateEvent'`
I looked into it and it looks like pghistory was recently upgraded to version 3.4.0. I am using that version (I did not change the pyproject.toml file). Any ideas for how to fix this?

**Comment by ERosendo** [comment, marked as answer]
Hi @v-anne, We recently updated `pghistory` to the latest version in this pull request: https://github.com/freelawproject/courtlistener/pull/4441

To make sure you're using the latest version, you can run this command:

```
docker compose up -d --build
```

This command will rebuild all your Docker images with the latest dependencies (thanks to the --build flag).

**Reply by v-anne** [reply]
Thank you @ERosendo, I will try that out.

**Reply by v-anne** [reply]
I'm now encountering a new error:
`ModuleNotFoundError: No module named 'environ'`

Running `pip show django-environ` in a shell of the cl-django container returns the following:

```pip show django-environ
Name: django-environ
Version: 0.11.2
Summary: A package that allows you to utilize 12factor inspired environment variables to configure your Django application.
Home-page: https://django-environ.readthedocs.org
Author: Daniele Faraglia
Author-email: daniele.faraglia@gmail.com
License: MIT
Location: /usr/local/lib/python3.12/site-packages
Requires: 
Required-by: 
```

**Reply by ERosendo** [reply]
@v-anne Which containers are showing the new error?

**Reply by ERosendo** [reply]
Are you using the code from the main branch? If not, please switch to the main branch and try again.

**Reply by v-anne** [reply]
All good now, @ERosendo. Thank you for your help.
